eng/doc/CrossPlatformCryptography.md

```diff
@@ -0,0 +1,336 @@
+| AES-GCM       | ✔️       | ✔️              |
+| DES-CBC       | ✔️       | ⚠️<sup>1</sup>  |
+| DES-ECB       | ✔️       | ⚠️<sup>1</sup>  |
+| 3DES-ECB      | ✔️       | ✔️              |
+| 3DES-CBC      | ✔️       | ✔️              |
+| RC4           | ✔️       | ⚠️<sup>1</sup>  |
+
+<sup>1</sup>When using OpenSSL 3, requires the legacy provider to be enabled.
+
+### AES-GCM keys, nonces, and tags
+
+* Key Sizes
+
+  AES-GCM works with 128, 192, and 256-bit keys.
+
+* Nonce Sizes
+
+  AES-GCM works with 12-byte nonces.
+
+* Tag Sizes
+  
+  AES-GCM works with 16-byte tags.
+
+## Asymmetric encryption
+
+This section includes the following subsections:
+
+* [RSA](#rsa)
+* [ECDSA](#ecdsa)
+* [ECDH](#ecdh)
+* [Ed25519](#ed25519)
+* [DSA](#dsa)
+
+### RSA
+
+This section includes the following packages:
+
+* [crypto/rsa](https://pkg.go.dev/crypto/rsa)
+
+| Padding Mode                           | Windows              | Linux               |
+|----------------------------------------|----------------------|---------------------|
+| OAEP (MD5)                             | ✔️                   | ✔️                 |
+| OAEP (SHA-1)                           | ✔️                   | ✔️                 |
+| OAEP (SHA-2)<sup>1</sup>               | ✔️                   | ✔️                 |
+| OAEP (SHA-3)                           | ❌                   | ❌                 |
+| PSS (MD5)                              | ✔️                   | ✔️                 |
+| PSS (SHA-1)                            | ✔️                   | ✔️                 |
+| PSS (SHA-2)<sup>1</sup>                | ✔️                   | ✔️                 |
+| PSS (SHA-3)                            | ❌                   | ❌                 |
+| PKCS1v15 Signature (Unhashed)          | ✔️                   | ✔️                 |
+| PKCS1v15 Signature (RIPMED160)         | ❌                   | ✔️<sup>2</sup>     |
+| PKCS1v15 Signature (MD4)               | ❌                   | ✔️<sup>2</sup>     |
+| PKCS1v15 Signature (MD5)               | ✔️                   | ✔️                 |
+| PKCS1v15 Signature (MD5-SHA1)          | ✔️<sup>2</sup>       | ✔️<sup>2</sup>     |
+| PKCS1v15 Signature (SHA-1)             | ✔️                   | ✔️                 |
+| PKCS1v15 Signature (SHA-2)<sup>1</sup> | ✔️                   | ✔️                 |
+| PKCS1v15 Signature (SHA-3)             | ❌                  | ❌                 |
+
+<sup>1</sup>The supported hash algorithms are the same as the ones supported as standalone hash functions.
+
+<sup>2</sup>Available starting in Microsoft Go 1.24.
+
+#### RSA key sizes
+
+[rsa.GenerateKey](https://pkg.go.dev/crypto/rsa#GenerateKey) only supports the following key sizes (in bits): 2048, 3072, 4096.
+
+Multi-prime RSA keys are not supported.
+
+The RSA key size is subject to the limitations of the underlying cryptographic library. For example, on Windows and when using SCOSSL, the key size should be multiple of 8.
```

Comment:
Is it worth being more exact? Are there more potential situations than this one example?

```suggestion
The RSA key size is subject to the limitations of the underlying cryptographic library:

* On Windows or when using SCOSSL, the key size must be a multiple of 8 (in bits).
```

But now I wonder if this was intended to be saying something about algorithms implemented by SCOSSL vs. by the built-in OpenSSL provider...

Reply:
I intentionally didn't want to enumerate a possible limitations, as can vary in time and are normally not part of the public documentation of the crypto libraries.

Reply:
I've rephrased the paragraph to be less confusing.